pulsar-function-go/pf/instanceConf_test.go

```diff
@@ -96,3 +98,18 @@ func TestInstanceConf_GetInstanceName(t *testing.T) {
 
 	assert.Equal(t, "101", instanceName)
 }
+
+func TestInstanceConf_Fail(t *testing.T) {
+	assert.Panics(t, func() {
+		newInstanceConfWithConf(&cfg.Conf{ProcessingGuarantees: 0, AutoACK: false})
+	}, "Should have a panic")
+	assert.Panics(t, func() {
+		newInstanceConfWithConf(&cfg.Conf{ProcessingGuarantees: 1, AutoACK: false})
+	}, "Should have a panic")
+	assert.Panics(t, func() {
+		newInstanceConfWithConf(&cfg.Conf{ProcessingGuarantees: 2})
+	}, "Should have a panic")
```

Comment:
Does this will introduce break change? If users upgrade to the new version but with the old configs.